libbeat/publisher/broker/brokertest/log.go

```diff
@@ -0,0 +1,136 @@
+package brokertest
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var debug bool
+var printLog bool
+
+type TestLogger struct {
+	t *testing.T
+}
+
+func init() {
+	flag.BoolVar(&debug, "debug", false, "enable test debug log")
+	flag.BoolVar(&printLog, "debug-print", false, "print test log messages right away")
+}
+
+type testLogWriter struct {
+	t *testing.T
+}
+
+func (w *testLogWriter) Write(p []byte) (int, error) {
+	w.t.Log(string(p))
+	return len(p), nil
+}
+
+func withLogOutput(fn func(*testing.T)) func(*testing.T) {
```

Comment:
originally I have had my own logger interface, to correctly capture log output within a test context (so log output and test name do make some more sense if something goes wrong). But when adopting pipeline to libbeat I opted for using logp. This enforced the introduction of withLogOutput, capturing stderr into `t.Log`, so I can still make sense of error logs mixed with pipeline processing logs.